vespalib/src/vespa/vespalib/hwaccelrated/iaccelrated.h

```diff
@@ -29,6 +30,9 @@ class IAccelrated
     virtual size_t populationCount(const uint64_t *a, size_t sz) const = 0;
     virtual double squaredEuclideanDistance(const float * a, const float * b, size_t sz) const = 0;
     virtual double squaredEuclideanDistance(const double * a, const double * b, size_t sz) const = 0;
+    // And 64 bytes from multiple sources
+    virtual void and64(size_t offset, const std::vector<std::pair<const uint64_t *, bool>> &src, uint64_t *dest) const = 0;
```

Comment:
these functions are a bit confusing. The length of the operation is specified in bytes while the offset is specified in multiples of uint64_t. It also binds the Word abstraction to be uint64_t, making that abstraction less useful. Also; at glace, the 64 in the function name seems to reflect the fact that we are using 64-bit values, leaving the size of the operation a mystery...

consider using byte offset and passing the memory as void*

Reply:
Agree, fixed